app/components/assets/notes/notes.tsx

```diff
@@ -1,18 +1,43 @@
-import { useLoaderData } from "@remix-run/react";
-
+import { useLoaderData, useFetcher } from "@remix-run/react";
+import { MarkdownViewer } from "~/components/markdown/markdown-viewer";
+import { useUserData } from "~/hooks";
+import { isFormProcessing } from "~/utils";
 import { NewNote } from "./new";
 import type { NoteWithDate } from "./note";
 import { Note } from "./note";
 
 export const Notes = () => {
   const { asset } = useLoaderData();
+  const user = useUserData();
   const hasNotes = asset?.notes.length > 0;
-
+  const fetcher = useFetcher();
+  let onSubmissionContent = "";
```

Comment:
Can you add a comment here with what is going on. The code is not immediately obvious so I think its important for people who look at this code to understand it.

Reply:
sure